app/src/main/java/com/example/drawingApp/ColorPickerUtil.kt

```diff
@@ -0,0 +1,209 @@
+package com.example.drawingApp
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.Color
+import android.util.Log
+import android.view.LayoutInflater
+import android.widget.TextView
+import androidx.core.widget.addTextChangedListener
+import com.google.android.material.bottomsheet.BottomSheetDialog
+import kotlin.math.max
+import kotlin.math.min
+
+object ColorPickerUtil {
+
+    data class ColorPack(
+        val hsv: Hsv,
+        val previousColor: Int
+    )
+
+    fun rgbToHsv(red: Int, green: Int, blue: Int): FloatArray {
+        val hsv = FloatArray(3)
+        Color.RGBToHSV(red, green, blue, hsv)
+        return hsv
+    }
+
+//    fun colorToHsv(color: Int): Hsv {
+//        val hsv = FloatArray(3)
+//        Color.colorToHSV(color, hsv)
+//        return Hsv(hsv[0], hsv[1], hsv[2])
+//    }
+
+    @SuppressLint("ClickableViewAccessibility")
+    fun colorPickerSheet(
+        context: Context,
+        /**
+         * Called whenever the hue is updated via text or slider.
+         * Use it to set wherever you are storing your hue
+         * and then update your color
+         */
+        onColorUpdate: (Hsv) -> Unit,
+        /**
+         * Called whenever text needs to be updated, meaning on any change, hsv, rgb, or text.
+         * Used to give the information of hsv and previous color.
+         * if there was no previous color pass it black. This is used to help
+         * populate the sheets text views and update the slider and gradient indicators as well as set the previous color box.
+         */
+        onColorTextForceUpdate: () -> ColorPack,
+        /**
+         * called whenever the dialog is dismissed or submitted
+         * on actual submission the color will not be null so take it in and set it wherever you need the selected color and set the hsv and rgb fields to defaults.
+         * on null don't do anything with it, don't set rgb and hsv to defaults.
+         */
+        onSubmission: (Int?) -> Unit
+    ) {
+        val MAX_HUE = 360f
```

Comment:
Don't make these local to a function, make them in the object

Reply:
fair enough, I tried to see if I could put const in the function and when it said no I just moved on and made it a val, I'll keep this in mind for the future